### PR TITLE
Feature/risk profile

### PR DESCRIPTION
**Pull request by jhughesbiot**
- Move all risk profiles methods to risk profile class
- Validate risk profile on creation and update
- Add risk profile unit tests

### CODE REVIEW

File: framework/python/src/common/risk_profile.py [hunk 1]
```diff
@@ -33,27 +33,29 @@ def __init__(self, profile_json=None, profile_format=None):
     if profile_json is None or profile_format is None:
       return
     self.name = profile_json['name']
-    self.created = datetime.now()
+    self.created = datetime.now().strftime('%Y-%m-%d')
     self.version = profile_json['version']
```
Comment (jhughesbiot) [inline framework/python/src/common/risk_profile.py:37]: Keep this as a timestamp, no need to convert to string outside o to_json method.

File: framework/python/src/common/risk_profile.py [hunk 2]
```diff
@@ -33,27 +33,29 @@ def __init__(self, profile_json=None, profile_format=None):
...
     self._validate(profile_json, profile_format)
     self._update_categories()
+    self._update_risk()
 
   # Load a profile without modifying the created date
   # but still validate the profile
   def load(self, profile_json, profile_format):
     self.name = profile_json['name']
-    self.created = datetime.strptime(profile_json['created'],
-                                     '%Y-%m-%d %H:%M:%S')
```
Comment (jhughesbiot) [inline framework/python/src/common/risk_profile.py]: Decode as a timestamp object so it doesn't break other methods

File: framework/python/src/common/risk_profile.py [hunk 3]
```diff
@@ -152,15 +165,16 @@ def _valid(self, profile_json, profile_format):
 
   def _expired(self):
     # Check expiry
-    created_date = self.created.timestamp()
+    created_date = datetime.strptime(
+      self.created, '%Y-%m-%d').timestamp()
```
Comment (jhughesbiot) [inline framework/python/src/common/risk_profile.py:169]: See note above about keeping this as a timestamp object, no need to decode string when kept as proper timestamp object

File: framework/python/src/common/risk_profile.py [hunk 4]
```diff
@@ -152,15 +165,16 @@ def _valid(self, profile_json, profile_format):
...
     today = datetime.now().timestamp()
     return created_date < (today - SECONDS_IN_YEAR)
 
   def to_json(self, pretty=False):
     json_dict = {
         'name': self.name,
         'version': self.version,
-        'created': str(self.created),
+        'created': self.created,
```
Comment (jhughesbiot) [inline framework/python/src/common/risk_profile.py:177]: See above comments concerning timestamp vs string object

File: framework/python/src/common/risk_profile.py (hunk 1 above)
Comment (jboddey) [reply framework/python/src/common/risk_profile.py:37]: The issue with using a timestamp, is that when it is passed to the UI, it passes it as a full timestamp in ISO format but I was wanting to pass it to the UI as a string in a particular format.

**Review by jboddey — APPROVED** [review]

**Pull request merged**